docs/design-decisions/ModelPromotion.md

```diff
@@ -0,0 +1,133 @@
+
+Data scientists can range from specialists in data analysis to proficient software engineers, or a blend of both skill sets.  The goal is to tailor the model development process to be as productive as possible when prototyping and experimenting as part of model development.  In the spirit of “fail fast”, implementing full engineering fundamentals on this experimentation process simply adds friction and may not be well-received by data scientists accustomed to performing model development on a stand-alone machine.  To meet these requirements a dedicated Prototyping repo with branch protection and a minimal CI workflow that includes linting and checks to prevent checking in secrets can provide the flexibility needed for model experimentation with light engineering fundamentals.
+
+Once a machine learning model under development by the data science team meets the target quality bar as defined by the customer, the model code in the Prototyping repo can be "promoted" into the Model Factory.
+
+## Model Promotion into the Model Factory
+
+Model promotion into the Model Factory is a collaborative process between the data scientist and the AI software engineer to incorporate the data scientists machine learning source artifacts into a production quality framework for maintaining, testing, and producing models for production use.
+
+Once models developed in the data science repo meet the requirements, they can be promoted.  During promotion, the model code, and automated tests are copied from the data science repo and integrated into the model framework.  Additionally, the dataset used to train the promoted model is integrated into existing ETL processes such that it is automatically registered in AML when ETL runs.
+
+The following diagram provides a high level overview of this process:
+
+![Model Lifecycle](../media/model-factory-workflow.png)
+
+For each model, a set of GitHub workflows are developed that help enforce software engineering fundamentals:
+
+![GitHub Workflows](../media/github-workflows.png)
+
+## Data Product and Dataset Management
+
+Management of models and the Datasets used to train models requires a production-quality approach to ensure traceability of a given model version to the dataset version used to train that model for production deployment.  At the same time, data scientists and software engineers require a low friction means to transform datasets as needed into new datasets to enable model training experimentation to meet performance goals.
```

Comment:
This can be a DataOps link as re-producibility and consistency in data pre-processing is a key.

Reply:
I reached out to him but have not heard back yet.  I will include this in another update later once I hear back.